packages/network-controller/tests/helpers.ts

```diff
@@ -192,6 +193,55 @@ export function buildMockGetNetworkClientById(
   return getNetworkClientById;
 }
 
+/**
+ * Builds a mock version of the `getNetworkClientIdByChainId` method on
+ * NetworkController.
+ *
+ * @param mockNetworkClientConfigurationsByNetworkClientId - Allows for defining
+ * the network client configuration — and thus the network client itself — that
+ * belongs to a particular network client ID.
+ * @returns The mock version of `getNetworkClientIdByChainId`.
+ */
+export function buildMockFindNetworkClientIdByChainId(
+  mockNetworkClientConfigurationsByNetworkClientId: Record<
```

Comment:
Would it be more valuable for this to be a set of network client configurations keyed by chain ID rather than network client ID? The reason why this was keyed by network client ID for `buildMockGetNetworkClientById` is because that's what `getNetworkClientById` returns, but in this case we might want something different.